src_MFsurfchem/MFsurfchem_functions.cpp

```diff
@@ -72,16 +72,38 @@ void init_surfcov(MultiFab& surfcov, const amrex::Real* dx)
         Dim3 hi = ubound(bx);
         const Array4<Real> & surfcov_arr = surfcov.array(mfi);
 
-        amrex::ParallelFor(bx, [=] AMREX_GPU_DEVICE (int i, int j, int k) noexcept
+        std::vector<amrex::Real> sum_surfcov0(n_ads_spec);
+        sum_surfcov0[0] = surfcov0[0];
+        for (int m=1;m<n_ads_spec;m++)
+            sum_surfcov0[m] = sum_surfcov0[m-1] + surfcov0[m];
+
+        amrex::ParallelForRNG(bx, [=] AMREX_GPU_DEVICE (int i, int j, int k, RandomEngine const& engine) noexcept
         {
-            if (k==0) {
-                for (int m=0;m<n_ads_spec;m++) {
-                    surfcov_arr(i,j,k,m) = surfcov0[m];
-                }
-            } else {
-                for (int m=0;m<n_ads_spec;m++) {
-                    surfcov_arr(i,j,k,m) = 0.;
+            if (k==0) 
+            {
+                amrex::Real Ntot = rint(surf_site_num_dens*dx[0]*dx[1]);  // total number of reactive sites
+                std::vector<int> Nocc(n_ads_spec);
```

Comment:
```suggestion
                GpuArray<int,MAX_SPECIES> Nocc;
```

Reply:
Why MAX_SPECIES? The size of the array containing the Number of occupied sites by each adsorption species should only be the number of adsorption species n_ads_spec, no?

Reply:
The size of GpuArrays must be specified at compile time - so putting in "variables" doesn't work.  MAX_SPECIES is #define'd to be 8 in common_namespace.H

There are ways around this but they are clunky.

Reply:
(although AMReX evolves over time, so it wouldn't surprise me if it's not so clunky to allocate space on the fly at this point)

Reply:
Since these two arrays are only used within init_surfcov as a way to sample the number of occupied sites, I think we probably do not need to have it the same size as MAX_SPECIES in the end and `GpuArray<int, n_ads_spec> Nocc` and `GpuArray<Real, n_ads_spec> sum_surfcov0` should suffice

Reply:
what your propose won't compile on the GPU, so you have to overallocate memory unfortunately and use MAX_SPECIES